justicecanada/otto
# perf: limit redundant files

This addresses ballooning storage issues with redundant file uploads and some technical debt re: unused files not being deleted. **Also adds more cleanup commands to Celery Beat, and tests them. Plus a few unrelated fixes to manage users CSV upload/download.**

* reuse SavedFile objects if possible to avoid duplicate files on disk
* delete SavedFile when there are no Document or ChatFile objects referencing it
* check hash client-side when file is uploaded in chat; if it exists on server, don't upload at all.

To test functionality:
* Delete all your chats and empty your media directory.
* Start a new chat.
* Drag a file in.
* Check the media directory for the file
* Drag the same file in (even with a different filename will work)
* Check the media directory - there should still be only 1 file.
* Both files should be in the library.
* Delete the chat
* Check the media directory - files should be gone.

You can also test from the Librarian modal which works similarly:
* Upload a file through Librarian modal.
* Check media directory
* Upload same file again.
* Check media directory - still should be just 1 instance of the file though there are now 2 documents
* Delete one of the documents in Librarian modal.
* File should still be there, since there is a reference from the other Document object.
* Delete the other Document
* File should be gone now.
